tests/compile-and-dump/Singletons/StandaloneDeriving.hs

```diff
@@ -0,0 +1,32 @@
+module Singletons.StandaloneDeriving where
+
+import Data.Singletons.Prelude
+import Data.Singletons.Prelude.Show
+import Data.Singletons.TH
+
+$(singletons [d|
+
+  infixl 6 :*:
+  data T a = a :*: a
+  data S = S1 | S2 deriving Eq
```

Comment:
The problem with `Eq` is that deriving `Eq` gets you `PEq` and `SEq`, *and* it gets you `SDecide`. That last one can't be defined in the `Partition` module, because it's not the result of promoting/singling, unlike all the other instances.